lib/chef/knife/ec2_base.rb

```diff
@@ -207,5 +219,25 @@ def ini_parse(file)
       end
       map
     end
+
+    # All valid platform
+    def valid_platform
+      ["ubuntu", "debian", "centos", "fedora", "rhel", "nginx", "turnkey", "jumpbox", "coreos", "cisco"]
```

Comment:
Okay, it looks like we could fix that, but not worth our time to spend too much time here.

https://github.com/fog/fog-aws/blob/master/lib/fog/aws/requests/compute/describe_images.rb